casingSimulations/sources.py

```diff
@@ -183,7 +177,8 @@ def plot(self, ax=None):
             )
         )
 
-    def validate(self):
+    @properties.validator
+    def _check_wire(self):
```

Comment:
Overriding `validate` may have unwanted behavior (eg `required` properties may not be checked anymore...). Instead, these should be registered as `validators`.

Only thing to be careful with is which methods override each other - not sure I got that right (eg `_check_wire` may be overridden by `_check_wire` in a subclass, but not `_check_wire_more`)